internal/common/hash.go

```diff
@@ -34,6 +35,12 @@ func BytesToHash(b []byte) Hash {
 	return h
 }
 
+// IsEmpty gets whether the hash contains all 0 bytes
+func (h Hash) IsEmpty() bool {
+	empty := Hash{}
```

Comment:
pull out as plain var, no need to make each time.

Reply:
Done.